runners/core-java/src/main/java/org/apache/beam/runners/core/ProcessFnRunner.java

```diff
@@ -77,7 +77,8 @@ public void finishBundle() {
 
   @Override
   public void onTimer(
-      String timerId, BoundedWindow window, Instant timestamp, TimeDomain timeDomain) {
+      String timerId, BoundedWindow window, Instant timestamp, Instant outputTimestamp,
```

Comment:
I would also change `timestamp` to `deliveryTimestamp` in all these methods, to make the difference very clear.